lib/_stream_readable.js

```diff
@@ -678,6 +677,17 @@ Readable.prototype.on = function(ev, fn) {
 };
 Readable.prototype.addListener = Readable.prototype.on;
 
+Readable.prototype.removeListener = function(type, listener) {
+  Stream.prototype.removeListener.call(this, type, listener);
+
+  if (type === 'readable' && this.listenerCount(type) === 0) {
+    const state = this._readableState;
+    state.readableListening = false;
+    state.emittedReadable = false;
+    state.needReadable = false;
```

Comment:
It seems like we should only be setting `readableListening` to `false` here, the rest are semaphore-ish flags.